ChangeLog.d/issue4313.txt

```diff
@@ -0,0 +1,25 @@
+Removals
+   * Remove the following macros: MBEDTLS_CHECK_PARAMS,
+     MBEDTLS_CHECK_PARAMS_ASSERT, MBEDTLS_PARAM_FAILED,
+     MBEDTLS_PARAM_FAILED_ALT, TEST_INVALID_PARAM, TEST_INVALID_PARAM_RET,
+     the following macros have been inactivated MBEDTLS_INTERNAL_VALIDATE_RET
+     and MBEDTLS_INTERNAL_VALIDATE, structures: param_failed_ctx_t,
+     mbedtls_test_param_failed_location_record_t, functions:
```

Comment:
Thanks for writing a thorough ChangeLog entry and making sure it is complete. However, I think test functions as well as internal functions and all.sh components don't need to be mentioned here as they are not part of the interface. The target audience for ChangeLog is mostly users of the library, so we should list everything that affects them, which is mostly just the public API of the library. Contributors/developers, who are interested in the tests and internal components, are expected to be able to figure some details on their own.

Reply:
I left only four macros from the beginning of the list.